semgrep-core/Parsing/pfff/json_to_generic.ml

```diff
@@ -13,19 +12,45 @@
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
  * license.txt for more details.
 *)
+open Common
 open Ast_json
+module M = Map_AST
 
 module G = AST_generic
 
+let expr x =
+  let e = Js_to_generic.expr x in
+
+  let visitor = M.mk_visitor {
+    M.default_visitor with
+    M.kexpr = (fun (k, _) e ->
+      (* apply on children *)
+      let e = k e in
+      match e with
+      | Record (lp, xs, rp) ->
+          let ys = xs |> List.map (function
+            | G.FieldStmt
+                ({s = G.DefStmt ({G.name = G.EN (G.Id (id, _)); _},
+                                 FieldDefColon ({vinit = Some e; _})); _}) ->
+                id, e
+            | _ -> failwith (spf "not a JSON field: %s" (G.show_expr e))
+          ) in
+          let zs = ys |> List.map (fun (id, e) ->
+            G.Tuple (G.fake_bracket [G.L (G.String id); e])) in
```

Comment:
Don't JSON records have real brackets we can use?

Reply:
I use them, see the use of lp and rp above; here it's for a single field, like foo: "bar", which we convert internally in a Tuple (we could use the token for ":" for the bracket ...).